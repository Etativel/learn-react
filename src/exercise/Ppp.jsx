import { useState } from "react";

export default function Ppp() {
  const [title, setTitle] = useState("react");

  function handleClick() {
    setTitle("yellow");
  }

  return (
    <div className="headline">
      <h1>{title}</h1>
      <p>{title}</p>
      <button onClick={handleClick}>Click me</button>
    </div>
  );
}
