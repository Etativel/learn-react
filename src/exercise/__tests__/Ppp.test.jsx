import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import userEvent from "@testing-library/user-event";
import Ppp from "../Ppp";

describe("App", () => {
  it("renders headline", () => {
    render(<Ppp />);

    screen.debug();

    expect(screen.getByRole("heading").textContent).toBe("react");
  });

  it("Ppp componenet", () => {
    const { container } = render(<Ppp />);

    expect(container).toMatchSnapshot();
  });

  it("user click", async () => {
    render(<Ppp />);

    // const button = screen.getByRole("button", { name: "Click me" });
    const button = screen.getByText("click me", { exact: false });

    await userEvent.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/yellow/i);
  });
});
