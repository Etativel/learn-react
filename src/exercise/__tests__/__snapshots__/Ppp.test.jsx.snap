// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Ppp componenet 1`] = `
<div>
  <div
    class="headline"
  >
    <h1>
      react
    </h1>
    <p>
      react
    </p>
    <button>
      Click me
    </button>
  </div>
</div>
`;
